src/nvim/main.c

```diff
@@ -394,7 +394,8 @@ int main(int argc, char **argv)
   bool use_remote_ui = (embedded_mode && !headless_mode);
   if (use_remote_ui) {
     TIME_MSG("waiting for UI");
-    remote_ui_wait_for_attach();
+    bool listen_and_embed = (embedded_mode && (params.listen_addr != NULL));
```

Comment:
`embedded_mode` is implied so only the `(params.listen_addr != NULL)` condition is necessary.

Reply:
cool, fixed.